### PR TITLE
London|ITP-Jan-2026|Ping Wang|Sprint 3|1-implement-and -rewrite -tests

### PR DESCRIPTION
**Pull request by pathywang**
- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

i did prep the do this coursework with node and npm test to check everything is ok

### CODE REVIEW

File: Sprint-3/1-implement-and-rewrite-tests/implement/1-get-angle-type.js [hunk 1]
```diff
@@ -35,3 +40,25 @@ function assertEquals(actualOutput, targetOutput) {
 // Example: Identify Right Angles
 const right = getAngleType(90);
 assertEquals(right, "Right angle");
+
+// Case 2: Identify Acute Angles:
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/implement/1-get-angle-type.js:46]: Can you think of a couple more edge cases that could happen here? Try to think of boundaries and possible invalid values.

File: Sprint-3/1-implement-and-rewrite-tests/implement/2-is-proper-fraction.js [hunk 2]
```diff
@@ -31,3 +31,27 @@ function assertEquals(actualOutput, targetOutput) {
 
 // Example: 1/2 is a proper fraction
 assertEquals(isProperFraction(1, 2), true);
+
+//Case 2: 5/5 is not a proper fraction
+assertEquals(isProperFraction(5,5), false);
+
+//Case 3: (-3)/(-5) is a proper fraction
+assertEquals(isFinite(-3,-5), true);
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/implement/2-is-proper-fraction.js]: Are we testing for the right thing here?

File: Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js [hunk 3]
```diff
@@ -22,8 +22,23 @@
...
+
+  
+  if (!validSuits.includes(suit) || !validRanks.includes(rank)) {
+    throw new Error("Invalid card");
+  }
+
+  if (rank === "A") return 11;
+  if (["J", "Q", "K"].includes(rank)) return 10;
+
+  return Number(rank);
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js:39]: Nice, I really like the use of the "guard pattern" here to deal with errors first 👍🏻.

File: Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js [hunk 4]
```diff
@@ -38,15 +53,44 @@ function assertEquals(actualOutput, targetOutput) {
...
 assertEquals(getCardValue("9♠"), 9);
+assertEquals(getCardValue("5♣"), 5);
+
+//Examples  A:
+assertEquals(getCardValue("A♦"), 11)
 
-// Handling invalid cards
-try {
-  getCardValue("invalid");
+//Examples J Q K:,
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js:64]: Seems like we might be missing a case here?

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js [hunk 5]
```diff
@@ -14,7 +14,24 @@ test(`should return "Acute angle" when (0 < angle < 90)`, () => {
...
+  expect(getAngleType(180)).toEqual("Straight angle");
+})
+
 // Case 5: Reflex angles
+test ('should return "Obtuse angle" when (180<angle && angle <360)',() => {
+  expect(getAngleType(275)).toEqual("Reflex angle");
+})
+
 // Case 6: Invalid angles
+test('should return "Invalid angle"when (angle>=360)', ()=> {
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js]: Again, let's focus on other edge cases and boundaries that could occur here!

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js [hunk 6]
```diff
@@ -4,7 +4,31 @@ const isProperFraction = require("../implement/2-is-proper-fraction");
...
 // TODO: Write tests in Jest syntax to cover all combinations of positives, negatives, zeros, and other categories.
 
-// Special case: numerator is zero
+/// Special case: denominator is zero
 test(`should return false when denominator is zero`, () => {
   expect(isProperFraction(1, 0)).toEqual(false);
 });
+
+//case:absolute numerator is bigger than absolute denominator
+test ('should return false when numerator absolute is bigger than denominator absolute',() => {
```
Comment (Lindronics) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js:13]: Are these all possible combinations? Keep in mind that you can group similar tests together with Jest.

**Review by Lindronics — COMMENTED** [review]
Overall looks great! I would add a few more assertions here and there to test for edge cases!

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
